discovery-infra/utils.py

```diff
@@ -267,6 +267,47 @@ def wait_till_at_least_one_host_is_in_status(
         raise
 
 
+def wait_till_at_least_one_host_is_in_stage(
+    client,
+    cluster_id,
+    stages,
+    timeout=consts.CLUSTER_INSTALLATION_TIMEOUT/2,
+    interval=5,
+):
+    log.info("Wait till 1 node is in stage %s" % stages)
+    try:
+        waiting.wait(
+            lambda: are_host_progress_in_stage(
+                client.get_cluster_hosts(cluster_id),
+                stages,
+            ),
+            timeout_seconds=timeout,
+            sleep_seconds=interval,
+            waiting_for="Node to be in of the stage %s" % stages,
+        )
+    except:
+        hosts = client.get_cluster_hosts(cluster_id)
+        log.info("All nodes: %s", hosts)
+        raise
+
+
+def are_host_progress_in_stage(hosts, stages, nodes_count=1):
+    log.info("Checking hosts installation stage")
+    hosts_in_stage = [host for host in hosts
+                      if (host["progress"]["current_stage"]) in stages]
+    if len(hosts_in_stage) >= nodes_count:
+        return True
+    host_info = [(host["id"], (host["progress"]["current_stage"])) for host in hosts]
+    log.info(
```

Comment:
`nodes_count` shouldn't be an integer with `%d`?
Anyway, use[ f-string](https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/) for readability.